source/matplot/backend/gnuplot.cpp

```diff
@@ -323,11 +321,9 @@ namespace matplot::backend {
         static std::string terminal_type;
         const bool dont_know_term_type = terminal_type.empty();
         if (dont_know_term_type) {
-            terminal_type =
-                run_and_get_output("gnuplot -e \"show terminal\" 2>&1");
-            terminal_type = std::regex_replace(
-                terminal_type, std::regex("[^]*terminal type is ([^ ]+)[^]*"),
-                "$1");
```

Comment:
@alandefreitas `std::regex` is ridiculously slow, consider `string_view`:
https://quick-bench.com/q/IUXYt_GIlS3VW8Gh4hJg8HLVQj4

Reply:
Yes. A custom parser would be better here, also in terms of compilation time.